nestkernel/exceptions.h

```diff
@@ -652,15 +652,13 @@ class DimensionMismatch : public KernelException
     : KernelException( "DimensionMismatch" )
     , expected_( -1 )
     , provided_( -1 )
-    , msg_( "" )
```

Comment:
There is no need to initialize the std::string, as it will get default-constructed and initialized to an empty string (https://cplusplus.com/reference/string/string/string/).

Reply:
@jougs What do you think about the explicit vs implicit initialization here? @JanVogelsang is right that there is no need, but with some members in the initializer list and others not, keeping track is more difficult. But then we have IDEs for keeping track ...

Reply:
I checked Stroustrup and in §17.4.1 he explicitly states that "[i]f a member constructor needs no argument, the member need not be mentioned in the member initializer list" and provides an example close to ours. Only built-in types definitely need initialization because they otherwise have undefined values. So I rest my case here.